runtime/lua/vim/uri.lua

```diff
@@ -35,31 +39,31 @@ do
   local function percent_encode_char(char)
     return '%' .. tohex(sbyte(char), 2)
   end
-  uri_encode = function(text, rfc)
-    if not text then
-      return
-    end
+
+  --- URI-encodes a string using percent escapes.
+  ---@param str string string to encode
+  ---@param rfc "rfc2396" | "rfc2732" | "rfc3986" | nil
+  ---@return string encoded string
+  function M.uri_encode(str, rfc)
     local pattern = PATTERNS[rfc] or PATTERNS.rfc3986
-    return text:gsub('([' .. pattern .. '])', percent_encode_char)
+    return (str:gsub('([' .. pattern .. '])', percent_encode_char))
```

Comment:
```suggestion
    return str:gsub('([' .. pattern .. '])', percent_encode_char)
```

Reply:
I put these extra parenthesis because `gsub()` returns multiple values on success, but only the 1st value is the result string.

> Returns a copy of s in which all (or the first n, if given) occurrences of the pattern have been replaced by a replacement string specified by repl, which can be a string, a table, or a function. gsub also returns, as its second value, the total number of matches that occurred.

Reply:
Ok good to know. That kind of thing can trip up anyone. Maybe worth adding a comment?